Add 1.2.0 release note.

jholdstock opened the pull request.

jholdstock commented on docs/release-note-1.2.0.md:
```diff
@@ -0,0 +1,508 @@
+Various Decred terminology has changed since the last release:
+
+- "Stakepools" are now known as "Voting Service Providers"
+- The name Decred should no longer styled as "deCRED"
```
Good catch. Fixed.

JoeGruffins commented on docs/release-note-1.2.0.md:
```diff
@@ -0,0 +1,508 @@
+This was done to address several security issues introduced by the default file storage.
+
+- A user who changes their password or email will now be logged out of all sessions on all devices.
+- Cookies stored in a browser can no longer be used obtain information after a user has logged out.
```
Totally my fault but
"used to obtain"

JoeGruffins commented on docs/release-note-1.2.0.md:
```diff
@@ -0,0 +1,508 @@
+|Config|PR|Reason|
+|------|--|------|
+|`wallethosts`, `walletusers`, `walletpasswords` and `walletcerts`|[#470](https://github.com/decred/dcrstakepool/pull/470)|dcrstakepool no longer contacts dcrwallet directly. All comms are now routed through stakepoold.|
+|`enablestakepoold`|[#398](https://github.com/decred/dcrstakepool/pull/398)|stakepoold is always required. dcrstakepool cannot functin without it.|
```
"function"

jholdstock commented:
Thanks Seth, Joe. I have addressed all of your comments.

I noticed Joe opened #510 which I think should definitely be included in the release, so let's finish that off and update this release note before merging it.

s-ben commented:
@jholdstock still seeing the Autoreconnect PR open and being worked on. I take it v1.2.0 is not released yet?

jholdstock commented:
Correct, 510 is the last piece to go in before release.

JoeGruffins commented:
I did an upgrade from the older version to the newer one following the guide.  I tried to go as far back as possible. The earliest Stakepoold I could make run without changes was around dcrd and dcrwallet release-v1.3, but I was unable to buy tickets.  Since all dcrstakepools are currently on release-v1.4 or are forked off the network, I think it is safe to start from there.  I started with a blank db and a new wallet. Dcrstakepool was on commit https://github.com/decred/dcrstakepool/commit/296ef81bf3a715b69fb04ae35a75b566f0a2a52a 
The following is evidence:
Starting home page.
![pic1](https://user-images.githubusercontent.com/34998433/64530273-a96ad380-d347-11e9-93db-2f22ec167ce8.png)

Made user and bought ticket.
![pic2](https://user-images.githubusercontent.com/34998433/64530577-72e18880-d348-11e9-901e-d5dc2e4cf8f1.png)

And the guide...

4. Perform an upgrade of each back-end server, one at a time
   i. Stop stakepoold
![pic3](https://user-images.githubusercontent.com/34998433/64530733-d4a1f280-d348-11e9-8552-6216c4ed6c7f.png)

   ii. Stop dcrwallet
![pic17](https://user-images.githubusercontent.com/34998433/64532983-88a57c80-d34d-11e9-8369-13be69c6159e.png)

   iii. Stop dcrd
![pic18](https://user-images.githubusercontent.com/34998433/64533048-ae328600-d34d-11e9-93a8-622a2a8f69ce.png)

   iv. Install latest dcrd binary and start
   (I just built)
![pic4](https://user-images.githubusercontent.com/34998433/64530980-5bef6600-d349-11e9-9c8a-b96956ea5900.png)

![pic9](https://user-images.githubusercontent.com/34998433/64531571-cf45a780-d34a-11e9-9453-18865424a85c.png)

   v. Install latest dcrwallet binary and start
![pic5](https://user-images.githubusercontent.com/34998433/64531130-b25ca480-d349-11e9-986c-69d517f0dd8c.png)

![pic10](https://user-images.githubusercontent.com/34998433/64531665-fa2ffb80-d34a-11e9-8447-1024a62df9a6.png)

   vi. Install latest stakepoold binary and start
![pic6](https://user-images.githubusercontent.com/34998433/64531236-f0f25f00-d349-11e9-85f0-6cea89f07df4.png)

![pic11](https://user-images.githubusercontent.com/34998433/64531732-1fbd0500-d34b-11e9-8640-e123ca00d4ec.png)

   vii. Check log files for warnings or errors
none

5. Make required changes to dcrstakepool.conf (detailed [above](#config-changes))
![pic12](https://user-images.githubusercontent.com/34998433/64531897-614db000-d34b-11e9-8933-0b18134ac874.png)

6. Install latest dcrstakepool and start
![pic13](https://user-images.githubusercontent.com/34998433/64532100-d1f4cc80-d34b-11e9-8d4e-e000e469f9c6.png)

![pic14](https://user-images.githubusercontent.com/34998433/64532189-fc468a00-d34b-11e9-9e4a-f0973154c8fd.png)

New homepage
![pic15](https://user-images.githubusercontent.com/34998433/64532273-21d39380-d34c-11e9-92d6-fc960c9566ed.png)

Ticket bought before updating
![pic16](https://user-images.githubusercontent.com/34998433/64532379-4fb8d800-d34c-11e9-914c-6ebfad197e04.png)

Conclustion: WORKS!

jholdstock commented:
This is excellent, thanks for going through that Joe. 

#510 and newly added #519 still need to go in before release.

According to a [report from @girino:decred.org](https://matrix.to/#/!XijfkJigmGCmMkgTTl:decred.org/$156797243524684yfzhv:decred.org?via=decred.org&via=matrix.org), the low fee tickets feature is not working correctly

jholdstock commented:
I have updated the release note with all recently merged issues. Just waiting for #527 to be merged and I think the release is ready to go :)
